corehq/apps/analytics/static/analytics/js/kissmetrics.js

```diff
@@ -0,0 +1,152 @@
+/* globals _, _kmq */
+
+var _kmq = _kmq || [];
+
+hqDefine('analytics/js/kissmetrics', function () {
+    'use strict';
+    var _get = hqImport('analytics/js/initial').getFn('kissmetrics'),
+        _abTests = hqImport('analytics/js/initial').getAbTests('kissmetrics'),
+        logger = hqImport('analytics/js/logging').getLoggerForApi('Kissmetrics'),
+        _utils = hqImport('analytics/js/utils'),
+        _init = {};
+
+    logger.verbose.addCategory('data', 'DATA');
+    logger.debug.addCategory('ab', 'AB TEST');
+
+    window.dataLayer = window.dataLayer || [];
+
+    var KmqWrapper = function (originalObject) {
+        Array.call(this, originalObject);
+    };
+    KmqWrapper.prototype = Object.create(Array.prototype);
+    KmqWrapper.prototype.constructor = KmqWrapper;
+    KmqWrapper.prototype.push = function () {
+        logger.deprecated.log(arguments, '_kmq.push');
+        Array.prototype.push.apply(this, arguments);
+    };
+    KmqWrapper.prototype.pushNew = function () {
+        Array.prototype.push.apply(this, arguments);
+    };
+    _kmq = new KmqWrapper(_kmq); // eslint-disable-line no-global-assign
+
+    /**
+     * Push data to _kmq by command type.
+     * @param {string} commandName
+     * @param {object} properties
+     * @param {function|undefined} callbackFn - optional
+     * @param {string|undefined} eventName - optional
+     */
+    var _kmqPushCommand = function (commandName, properties, callbackFn, eventName) {
+        var command, data;
+        command = _.filter([commandName, eventName, properties, callbackFn], function (a) {
+            return !_.isUndefined(a);
+        });
+        _kmq.pushNew(command);
```

Comment:
What difference does it make if the undefined arguments are filtered out before calling `pushNew`?